test/lib/command_test.rb

```diff
@@ -27,46 +27,78 @@ def test_executes_block_passed_to_command
     @truthy = false
     NoArgCommand.call do |r, c|
       @truthy = true
+      r.pass; r.fail
     end
 
     assert_equal true, @truthy
   end
 
+  def test_executes_pass_block_on_pass
+    @truthy = false
+    result = ArgCommand.call(arg: "arg") {|r|
+      r.pass {
+        @truthy = true
+      }
+      r.fail {
+        flunk "Should not have called the fail block on a pass"
+      }
+    }
+
+    assert_equal true, @truthy
+  end
+
+  # TODO: Refactor into an "assert_executes" test helper
+  def test_executes_fail_block_on_fail
+    @truthy = false
+    result = ArgCommand.call(arg: nil) {|r|
+      r.pass {
+        flunk "Should not have called the fail block on a pass"
+      }
+      r.fail {
+        @truthy = true
+      }
+    }
+
+    assert_equal true, @truthy
+  end
+
   def test_result_and_command_available_in_block
     NoArgCommand.call do |r, c|
       assert c.is_a? NoArgCommand
 
       assert r.is_a? Slayer::Result
       assert_equal true, r.success?
+
+      r.pass; r.fail
```

Comment:
Should we include a third possibility, `r.skip`, which allows you to do nothing with the response object?